### ISSUE TITLE
False positive of `regexp/no-unused-capturing-group`

### ISSUE THREAD

**Issue by zloirock** [ota-meshi/eslint-plugin-regexp#353]
```js
const string = 'foo:abc,bar:def'.replace(RegExp('foo:(?<foo>\\w+),bar:(?<bar>\\w+)'), (...args) => {
  const { foo, bar } = args.pop();
  return `${ bar },${ foo }`;
});
// =>
// error  Capturing group 'foo' is defined but never used  regexp/no-unused-capturing-group
// error  Capturing group 'bar' is defined but never used  regexp/no-unused-capturing-group
```

**Comment by RunDevelopment** [comment]
Additional info: It seems like it doesn't matter what you with `args` at all.

```js
const string = 'foo:abc,bar:def'.replace(
  /foo:(?<foo>\\w+),bar:(?<bar>\\w+)/,
  (...args) => args.join("")
);
```

![image](https://user-images.githubusercontent.com/20878432/136262794-942e46f0-4e76-4057-b672-88f410de40e1.png)
